src/core/Proc/Async.pm6

```diff
@@ -412,15 +424,24 @@ my class Proc::Async {
     proto method kill(|) {*}
     multi method kill(Proc::Async:D: Signal:D \signal = SIGHUP) {
         X::Proc::Async::MustBeStarted.new(:method<kill>, proc => self).throw if !$!started;
+        $!stdout_supplier.done() if $!stdout_supplier;
```

Comment:
Will users expect `&done` or `&quit` callbacks to be fired? I'm not really sure.

Reply:
Currently, the ```killprocasync``` op invoked in the method call always cancels the work on the event queue ([in MoarVM at least](https://github.com/MoarVM/MoarVM/blob/master/src/io/procops.c#L1007)). However, not knowing that, it isn't unreasonable that the user might expect that the process could recover from the signal.

I was originally approaching it from the standpoint that the user is going in expecting the termination of the process, but I'm not so sure now. Might be too late to change the behavior though?